### PR TITLE
Migrate travis tests to Github Actions

### PR DESCRIPTION
**Pull request by ljharb**
See #127

~~2 pnpm tests [are failing](https://github.com/ljharb/install-peerdeps/runs/2255110290?check_suite_focus=true), but pass for me locally, so I'm not sure how to best debug this.~~

### CODE REVIEW

**Comment by ljharb** [comment]
I'm not sure why the actions aren't running here; @nathanhleung, you may need to check a box in the repo settings to allow all actions to run?

**Comment by karlhorky** [comment]
> 2 pnpm tests [are failing](https://github.com/ljharb/install-peerdeps/runs/2255110290?check_suite_focus=true), but pass for me locally, so I'm not sure how to best debug this.

Taking a look at these now, I'll see what I can find out by running locally. Maybe I will need to fork your fork to test on GitHub Actions.

**Comment by karlhorky** [comment]
Ok, after showing `stdout` and `stderr`, it seems `stderr` could be somewhat helpful here:

```
stdout: ERR spawn pnpm ENOENT
```

Maybe it's having trouble finding `pnpm` for some reason (`ENOENT` is "file or directory not found"). Not sure if it's related to https://github.com/nathanhleung/install-peerdeps/issues/123 and the original commit https://github.com/nathanhleung/install-peerdeps/commit/1b386d5015e17309162e8e6bb419bf73b2f27ac9

Debugging more...

**Comment by karlhorky** [comment]
`$PATH` seems to be fine:

```
 npm config get prefix: /usr/local

 $PATH:  /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/runner/work/install-peerdeps/install-peerdeps/node_modules/.bin:/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/runner/work/install-peerdeps/install-peerdeps/node_modules/.bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/runner/.local/bin:/opt/pipx_bin:/usr/share/rust/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

But the `pnpm` binary is not in `/usr/local/bin` 🤔 (and also no `yarn`) Maybe the `pnpm` and `yarn` installation is not working as expected... Continuing debugging...

```
  ls: total 916192
  drwxr-xr-x  2 root   root        4096 Mar 30 06:17 .
  drwxr-xr-x 18 root   root        4096 Mar 30 06:04 ..
  -rw-r--r--  1 root   root         954 Mar 20 23:44 README.md
  -rwxr-xr-x  1 root   root        3459 Mar 30 06:04 activate-global-python-argcomplete
  -rwxr-xr-x  1 root   root    32792160 Feb 19 11:18 aliyun
  -rwxr-xr-x  1 runner docker  25882983 Feb 26 23:23 azcopy
  -rwxr-xr-x  1 root   root         539 Mar 30 05:39 bin-proxy
  -rwxr-xr-x  1 root   root         529 Mar 30 05:37 bundle
  -rwxr-xr-x  1 root   root         531 Mar 30 05:37 bundler
  -rwxr-xr-x  1 root   root    12682744 Mar 23 15:54 ccmake
  -rwxr-xr-x  1 root   root    12727032 Mar 23 15:54 cmake
  -rwxr-xr-x  1 root   root    28152448 Mar 23 15:54 cmake-gui
  -rwxr-xr-x  1 root   root    13177080 Mar 23 15:54 cpack
  -rwxr-xr-x  1 root   root    14142136 Mar 23 15:54 ctest
  -rwxr-xr-x  1 root   root    12212176 Mar 30 03:11 docker-compose
  -rwxr-xr-x  1 root   root         525 Mar 30 05:39 dotenv
  -rwxr-xr-x  1 root   root         537 Mar 30 05:39 fastlane
  -rwxr-xr-x  1 root   root         581 Mar 30 05:39 generate-api
  -rwxr-xr-x  1 root   root    39895040 Mar 30 05:20 helm
  -rwxr-xr-x  1 root   root         549 Mar 30 05:39 httpclient
  -rwxr-xr-x  1 runner docker   9814937 Mar  5  2020 hub
  -rwxr-xr-x  1 root   root         507 Mar 30 05:37 irb
  -rwxr-xr-x  1 root   root     7452113 Mar 30 05:20 kind
  -rwxr-xr-x  2 root   root    72732776 Mar 20 23:44 kubectl
  -rwxr-xr-x  1 root   root    13406208 Mar 30 05:20 kustomize
  -rwxr-xr-x  1 root   root       13541 Mar 30 05:20 lein
  -rwxr-xr-x  1 root   root    59794699 Mar 30 05:20 minikube
  -rwxr-xr-x  1 root   root    73873984 Mar 30 05:28 node
  -rwxr-xr-x  2 root   root    72732776 Mar 20 23:44 oc
  -rwxr-xr-x  1 runner docker  10657733 Mar 18 17:52 oras
  -rwxr-xr-x  1 root   root   129019904 Feb 17 11:15 packer
  -rwxr-xr-x  1 root   root     3082851 Mar 17 07:29 phpunit
  -rwxr-xr-x  1 root   root         206 Mar 30 06:04 pipx
  -rwxr-xr-x  1    501 staff   56919325 Mar 25 19:04 pulumi
  -rwxr-xr-x  1    501 staff        245 Mar 25 19:02 pulumi-analyzer-policy
  -rwxr-xr-x  1    501 staff       1853 Mar 25 19:02 pulumi-analyzer-policy-python
  -rwxr-xr-x  1    501 staff   17063638 Mar 25 19:06 pulumi-language-dotnet
  -rwxr-xr-x  1    501 staff   31352558 Mar 25 19:07 pulumi-language-go
  -rwxr-xr-x  1    501 staff   21185369 Mar 25 19:06 pulumi-language-nodejs
  -rwxr-xr-x  1    501 staff   21117593 Mar 25 19:06 pulumi-language-python
  -rwxr-xr-x  1    501 staff       4724 Mar 25 19:02 pulumi-language-python-exec
  -rwxr-xr-x  1    501 staff        238 Mar 25 19:02 pulumi-resource-pulumi-nodejs
  -rwxr-xr-x  1    501 staff       1893 Mar 25 19:02 pulumi-resource-pulumi-python
  -rwxr-xr-x  1 root   root        2555 Mar 30 06:04 python-argcomplete-check-easy-install-script
  -rwxr-xr-x  1 root   root         314 Mar 30 06:04 python-argcomplete-tcsh
  -rwxr-xr-x  1 root   root         513 Mar 30 05:38 racc
  -rwxr-xr-x  1 root   root         513 Mar 30 05:38 rake
  -rwxr-xr-x  1 root   root         513 Mar 30 05:38 rdoc
  -rwxr-xr-x  1 root   root      943940 Feb 25 01:05 rebar3
  -rwxr-xr-x  1 root   root        1917 Mar 30 06:04 register-python-argcomplete
  -rwxr-xr-x  1 root   root         509 Mar 30 05:38 ri
  -rwxr-xr-x  1 root   root         523 Mar 30 05:39 rougify
  -rwxr-xr-x  1 root   root    62414064 Mar 30 05:17 stack
  -rwxr-xr-x  1 root   root         591 Mar 30 05:39 terminal-notifier
  -rwxr-xr-x  1 root   root    82741780 Mar 24 18:42 terraform
  -rwxr-xr-x  1 root   root         219 Mar 30 06:04 userpath
  -rwxr-xr-x  1 root   root         529 Mar 30 05:39 ww
  -rwxr-xr-x  1 root   root         543 Mar 30 05:39 xcodeproj
  -rwxr-xr-x  1 root   root         537 Mar 30 05:39 xcpretty
  -rwxr-xr-x  1 root   root         639 Mar 30 05:39 xcpretty-travis-formatter
```

**Comment by karlhorky** [comment]
Ok, got it working: https://github.com/karlhorky/install-peerdeps/runs/2259745553?check_suite_focus=true

It seems to be a problem with the `ljharb/actions/node/install@main` action and the `before_install: npm install -g pnpm@3` property - it needs `sudo` there.

Adding a suggestion to the PR now.

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      fail-fast: false
+      matrix: ${{ fromJson(needs.matrix.outputs.latest) }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ljharb/actions/node/install@main
+        name: 'nvm install ${{ matrix.node-version }} && npm install'
+        with:
+          node-version: ${{ matrix.node-version }}
+          before_install: npm install -g pnpm@3 yarn@1
```
Comment (karlhorky) [inline .github/workflows/node.yml:33]:
```suggestion
          before_install: sudo npm install -g pnpm@3 yarn@1
```

**Comment by karlhorky** [comment]
Maybe a suggestion for the `ljharb/actions/node/install@main` action would be: fail if the `before_install` command errors out.

File: .github/workflows/node.yml (hunk 1 above)
Comment (ljharb) [reply .github/workflows/node.yml:33]: ahhh thank you

**Comment by ljharb** [comment]
yes, thanks, it should already have failed on that; it's a bug that it doesn't. i'll update the action, then the PR.

**Comment by ljharb** [comment]
[although](https://github.com/ljharb/install-peerdeps/runs/2255110290?check_suite_focus=true):
```
******> npm install -g pnpm@3 yarn@1

added 2 packages, and audited 3 packages in 3s
```
which suggests it did in fact succeed. maybe it's a username thing.

**Comment by ljharb** [comment]
I added some debug info to the action, and it turns out that's the issue (it's using nvm, so sudo shouldn't be required). I'll investigate a bit more.

**Comment by ljharb** [comment]
ok, fixed the action, so now it should work without changes :-)

**Review by nathanhleung — APPROVED** [review]
Thanks @ljharb!!

**Comment by ljharb** [comment]
@nathanhleung before merging this, did you enable actions to run on this repo? the actions never actually ran on this PR.

**Comment by karlhorky** [comment]
Seems like they're running now (at least on `master`):

Screen Shot 2021-04-06 at 10 21 39

Maybe it has something to do with the restrictions that GitHub Actions have on running actions from forks?

**Comment by nathanhleung** [comment]
> @nathanhleung before merging this, did you enable actions to run on this repo? the actions never actually ran on this PR.

Screen Shot 2021-04-06 at 10 24 33 AM

These are the actions settings I'm seeing for the repo, not sure if there's anything else I needed to enable? In any case, I see that they've run on my end also.

> Maybe the reason they didn't run in this PR has something to do with the restrictions that GitHub Actions have on running actions from forks?

I think I remember reading something about this somewhere, so this sounds right.

Thank you to both of you for coordinating this effort!

**Comment by ljharb** [comment]
hmm, actions from forks usually run for me, but maybe that’s it. Thanks for checking :-)

**Pull request merged**